### ISSUE TITLE
build error on exploit.c: redefinition of 'struct mmsghdr'

### ISSUE THREAD

**Issue by passos** [android-rooting-tools/android_run_root_shell#30]
SimonMacBookPro:android_run_root_shell simon$ ndk-build NDK_PROJECT_PATH=. APP_BUILD_SCRIPT=./Android.mk
Android NDK: WARNING:libexploit/Android.mk:exploit: LOCAL_LDFLAGS is always ignored for static libraries
[armeabi] Compile thumb  : run_root_shell <= cred.c
[armeabi] Compile thumb  : run_root_shell <= kallsyms.c
[armeabi] Compile thumb  : run_root_shell <= main.c
[armeabi] Compile thumb  : run_root_shell <= mm.c
[armeabi] Compile thumb  : run_root_shell <= ptmx.c
[armeabi] Compile thumb  : exploit <= exploit.c
libexploit/exploit.c: In function 'find_kernel_text_from_iomem':
libexploit/exploit.c:114:5: warning: format '%x' expects argument of type 'unsigned int', but argument 2 has type 'long unsigned int' [-Wformat=]
     printf("Detected kernel physical address at 0x%08x from iomem\n", kernel_ram);
     ^
[armeabi] Compile thumb  : exploit <= mmap.c
[armeabi] StaticLibrary  : libexploit.a
[armeabi] Compile thumb  : kallsyms <= kallsyms_in_memory.c
[armeabi] StaticLibrary  : libkallsyms.a
[armeabi] Compile thumb  : diagexploit <= diag.c
[armeabi] StaticLibrary  : libdiagexploit.a
[armeabi] Compile thumb  : perf_event_exploit <= perf_event.c
[armeabi] Compile thumb  : perf_event_exploit <= perf_swevent.c
[armeabi] StaticLibrary  : libperf_event_exploit.a
[armeabi] Compile thumb  : msm_acdb_exploit <= acdb.c
[armeabi] StaticLibrary  : libmsm_acdb_exploit.a
[armeabi] Compile thumb  : fj_hdcp_exploit <= fj_hdcp.c
[armeabi] StaticLibrary  : libfj_hdcp_exploit.a
[armeabi] Compile thumb  : fb_mem_exploit <= fb_mem.c
[armeabi] StaticLibrary  : libfb_mem_exploit.a
[armeabi] Compile thumb  : msm_cameraconfig_exploit <= msm_cameraconfig.c
[armeabi] StaticLibrary  : libmsm_cameraconfig_exploit.a
[armeabi] Compile thumb  : put_user_exploit <= put_user.c
[armeabi] StaticLibrary  : libput_user_exploit.a
[armeabi] Compile thumb  : get_user_exploit <= get_user.c
[armeabi] StaticLibrary  : libget_user_exploit.a
[armeabi] Compile thumb  : futex_exploit <= futex.c
[armeabi] Compile thumb  : futex_exploit <= exploit.c
libexploit/libfutex_exploit/exploit.c:82:8: error: redefinition of 'struct mmsghdr'
 struct mmsghdr {
        ^
In file included from libexploit/libfutex_exploit/exploit.c:24:0:
/opt/android-ndk-r10b/platforms/android-L/arch-arm/usr/include/sys/socket.h:99:8: note: originally defined here
 struct mmsghdr {
        ^
make: **\* [obj/local/armeabi/objs/futex_exploit/exploit.o] Error 1

**Comment by iamalsaher** [comment]
@passos Did you find a solution? I am having same type of error for libexploits, can you help

**Comment by passos** [comment]
@iamalsaher no. It's too long ago. I don't even remember why I built this project.

**Comment by DARKTEMPLE9** [comment]
any idea to solve this problem?

**Comment by iamalsaher** [comment]
Nope nothing

**Comment by dadreamer** [comment]
Change it to be _mmsghdr**2**_ (for instance) everywhere in the code and the compilation will go on.
